src/bin/miri.rs

```diff
@@ -60,6 +60,12 @@ impl rustc_driver::Callbacks for MiriCompilerCalls {
 
         queries.global_ctxt().unwrap().peek_mut().enter(|tcx| {
             init_late_loggers(tcx);
+            if !tcx.sess.opts.crate_types.contains(&CrateType::Executable) {
```

Comment:
Oh looks like this returns false by default...??

Reply:
Ah no the problem is using `opts.crate_types` rather than `crate_types()`. That's a strange footgun?